core/lib/dal/README.md

```diff
@@ -3,52 +3,117 @@
-| proof                                 |
-| protective_reads                      |
-| prover_jobs                           |
-| static_artifact_storage               |
-| storage_logs_dedup                    |
-| tokens                                |
-| transaction_traces                    |
-| transactions                          |
+Current schema is managed by `sqlx`. Schema changes are stored in the [`migrations`](migrations) directory.
+
+## Schema overview
+
+_This overview skips prover-related and Ethereum sender-related tables, which are specific to the main node._
+
+### Miniblocks and L1 batches
+
+- `miniblocks`. Stores miniblock headers.
+
+- `miniblocks_consensus`. Stores miniblock data related to the consensus algorithm used by the decentralized sequencer.
+  Tied one-to-one to miniblocks (the consensus side of the relation is optional).
+
+- `l1_batches`. Stores L1 batch headers.
+
+- `commitments`. Stores a part of L1 batch commitment data (event queue and bootloader memory commitments). In the
+  future, other commitment-related columns will be moved here from `l1_batches`.
+
+### Transactions
+
+- `transactions`. Stores all transactions received by the node, both L2 and L1 ones. Transactions in this table are not
+  necessarily included into a miniblock; i.e., the table is used as a persistent mempool as well.
+
+### VM storage
+
+See [`zksync_state`] crate for more context.
+
+- `storage_logs`. Stores all the VM storage write logs for all transactions, as well as non-transaction writes generated
+  by the bootloader. This is the source of truth for the VM storage; all other VM storage implementations (see the
+  [`zksync_state`] crate) are based on it (e.g., by adding persistent or in-memory caching). Used by multiple components
+  including Metadata calculator, Commitment generator, API server (both for reading one-off values like account balance,
+  and as a part of the VM sandbox) etc.
+
+- `initial_writes`. Stores initial writes information for each L1 batch, i.e., the enumeration index assigned for each
+  key. Used when generating L1 batch metadata in Metadata calculator and Commitment generator components.
+
+- `protective_reads`. Stores protective read information for each L1 batch, i.e., keys influencing VM execution for the
+  batch that were not modified. Used when generating L1 batch metadata in Commitment generator.
+
+- `factory_deps`. Stores bytecodes of all deployed L2 contracts.
+
+- `storage`. **Obsolete, going to be removed; must not be used in new code.**
+
+### Other VM artifacts
+
+- `events`. Stores all events (aka logs) emitted by smart contracts during VM execution.
+
+- `l2_to_l1_logs`. Stores L2-to-L1 logs emitted by smart contracts during VM execution.
+
+- `call_traces`. Stores call traces for transactions emitted during VM execution. (Unlike with L1 node implementations,
+  in Era call traces must be proactively generated for all transactions to function correctly.)
```

Comment:
I'm not sure about "must". They're generated proactively because otherwise it's needed to rerun the whole batch to get them which is a heavy operation but possible.